rpc/handlers_test.go

```diff
@@ -1940,6 +1940,58 @@ func TestTransactionStatus(t *testing.T) {
 	}
 }
 
+func TestTransactionNotFoundStatus(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	tests := []struct {
+		network        utils.Network
+		notFoundTxHash *felt.Felt
+	}{
+		{
+			network:        utils.MAINNET,
+			notFoundTxHash: utils.HexToFelt(t, "0x8c96a2b3d73294667e489bf8904c6aa7c334e38e24ad5a721c7e04439ff9"),
+		},
+		{
+			network:        utils.INTEGRATION,
+			notFoundTxHash: utils.HexToFelt(t, "0xd7747f3d0ce84b3a19b05b987a782beac22c54e66773303e94ea78cc3c15"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.network.String(), func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			t.Cleanup(mockCtrl.Finish)
+
+			t.Run("transaction not found", func(t *testing.T) {
+				notFoundTests := map[string]struct {
+					finality      rpc.TxnFinalityStatus
+					hash          *felt.Felt
+					expectedError *jsonrpc.Error
+				}{
+					"not found": {
+						hash: test.notFoundTxHash,
+					},
+				}
```

Comment:
There appears to be some unnecessary code here and below. I think we can have just one `for-range` through `tests`, no need for nested loops.

Reply:
done